docs/examples/VALIDATOR.md

```diff
@@ -0,0 +1,126 @@
+Examples of validators and minting policies written in Plutarch.
+
+> Note: If you spot any mistakes/have any related questions that this guide lacks the answer to, please don't hesitate to raise an issue. The goal is to have high quality documentation for Plutarch users!
+
+- [Validator that always succeeds](#validator-that-always-succeeds)
+- [Validator that always fails](#validator-that-always-fails)
+- [Validator that checks whether a value is present within signatories](#validator-that-checks-whether-a-value-is-present-within-signatories)
+- [Using custom datum/redeemer in your Validator](#using-custom-datumredeemer-in-your-validator)
+
+> Aside: Be sure to check out [Compiling and Running](./../GUIDE.md#compiling-and-running) first!
+
+# Validator that always succeeds
+
+```hs
+import Plutarch.Prelude
+import Plutarch.Api.V1.Contexts
+import Plutarch.Api.V1.Scripts
+
+alwaysSucceeds :: Term s (PAsData PDatum :--> PAsData PRedeemer :--> PAsData PScriptContext :--> PUnit)
+alwaysSucceeds = plam $ \datm redm ctx -> pconstant ()
+```
+
+All the arguments are ignored. So we use the generic `PDatum` and `PRedeemer` types.
+
+Execution:
+
+```hs
+import qualified PlutusTx
+
+> alwaysSucceeds `evalWithArgsT` [PlutusTx.toData (), PlutusTx.toData (), PlutusTx.toData ()]
+Right (Program () (Version () 1 0 0) (Constant () (Some (ValueOf unit ()))))
+```
+
+# Validator that always fails
+
+```hs
+import Plutarch.Prelude
+import Plutarch.Api.V1.Contexts
+import Plutarch.Api.V1.Scripts
+
+alwaysFails :: Term s (PAsData PDatum :--> PAsData PRedeemer :--> PAsData PScriptContext :--> PUnit)
```

Comment:
It should be `PData :--> PData :--> PAsData PScriptContext :--> POpaque`

Reply:
When are the `PDatum` and `PRedeemer` types supposed to be used? Also why use `POpaque` in a concrete validator function? I think that should really be necessary when you wrap the validator function to get a `PValidator`.